src/test/pull_hf_model_test.cpp

```diff
@@ -137,30 +139,43 @@ const std::string expectedGraphContents = R"(
 
 TEST_F(HfDownloaderPullHfModel, PositiveDownload) {
     std::string modelName = "OpenVINO/Phi-3-mini-FastDraft-50M-int8-ov";
-    std::string downloadPath = ovms::FileSystem::appendSlash(this->directoryPath) + "repository";
-    this->ServerPullHfModel(modelName, downloadPath);
-    std::string modelPath = ovms::FileSystem::appendSlash(downloadPath) + "openvino_model.bin";
-    std::string graphPath = ovms::FileSystem::appendSlash(downloadPath) + "graph.pbtxt";
-    ASSERT_EQ(std::filesystem::exists(modelPath), true);
-    ASSERT_EQ(std::filesystem::exists(graphPath), true);
+    std::string downloadPath = ovms::FileSystem::joinPath({this->directoryPath, "repository"});
+    std::string task = "text_generation";
+    this->ServerPullHfModel(modelName, downloadPath, task);
+
+    std::string basePath = ovms::FileSystem::joinPath({this->directoryPath, "repository", "OpenVINO", "Phi-3-mini-FastDraft-50M-int8-ov"});
+    std::string modelPath = ovms::FileSystem::appendSlash(basePath) + "openvino_model.bin";
+    std::string graphPath = ovms::FileSystem::appendSlash(basePath) + "graph.pbtxt";
+
+    ASSERT_EQ(std::filesystem::exists(modelPath), true) << modelPath;
+    ASSERT_EQ(std::filesystem::exists(graphPath), true) << graphPath;
     ASSERT_EQ(std::filesystem::file_size(modelPath), 52417240);
     std::string graphContents = GetFileContents(graphPath);
     std::cout << graphContents << std::endl;
 
     ASSERT_EQ(expectedGraphContents, graphContents);
 }
 
+TEST_F(HfDownloaderPullHfModel, NegativeNoDownloadWrongTask) {
+    std::string modelName = "OpenVINO/Phi-3-mini-FastDraft-50M-int8-ov";
+    std::string downloadPath = ovms::FileSystem::joinPath({this->directoryPath, "repository"});
+    std::string task = "1";
```

Comment:
I don't think we need to waste time on server setup for validation of those parameters. Couldn't we test it at lower level?